test/cli/test_log_runtime_archive_main.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+require "syskit/test/self"
+require "syskit/cli/log_runtime_archive_main"
+
+module Syskit
+    module CLI
+        describe CLIArchiveMain do
+            before do
+                @root = make_tmppath
+                @archive_dir = make_tmppath
+                @mocked_files_sizes = []
+
+                10.times { |i| (@archive_dir / i.to_s).write(i.to_s) }
+
+                @archiver = LogRuntimeArchive.new(@root, @archive_dir)
+            end
+
+            it "removes enough files to reach the freed limit" do
+                size_files = [6, 2, 1, 6, 7, 10, 3, 5, 8, 9]
+                mock_files_size(size_files)
+                mock_available_space(0.5)
+
+                call_command_line(@root, @archive_dir, 1e-3, 10 * 1e-3)
```

Comment:
This is the test's bug.

1e-3 MB is 1 kB. It should have removed all files. You've "cheated" by setting mock_available_space to 0.5, which does not exist (mock_available_space takes bytes as input)

Reply:
fixed